Address CI failures in the workflow "Install from conda-forge"

The job "Check" within the "Install from conda-forge" GitHub Action workflow fails with the following error message:

https://github.com/iiasa/message_ix/runs/3812369811?check_suite_focus=true

```
message-ix: D:\a\_temp\bdcbacfc-6fdb-4503-8522-b897f70c34fa.ps1:6
Line |
   6 |  message-ix show-versions
     |  ~~~~~~~~~~
     | The term 'message-ix' is not recognized as a name of a cmdlet, function, script file, or executable
     | program. Check the spelling of the name, or if a path was included, verify that the path is correct
     | and try again.

Error: Process completed with exit code 1.
```

Looking at the packages in the job "Install message-ix", message-ix as well as ixmp are not installed in their latest version.

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -83,7 +83,7 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
-        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh
+        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh  || true
         conda activate ${{ env.CONDA_ENV }}
 
         conda info

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -56,6 +56,9 @@ jobs:
         installer: ${{ matrix.conda }}
         version: py39_4.9.2
 
+    - name: Configure PowerShell for "conda activate"
+      run: conda init powershell
+
     - name: Install message-ix
       if: ${{ matrix.os == 'macos-latest' }}
       run: |

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -18,7 +18,8 @@ jobs:
   conda:
     strategy:
       matrix:
-        os: [windows-latest, macos-latest]
+        # os: [windows-latest, macos-latest]
+        os: [macos-latest]
         conda: [anaconda, miniconda]
         extra_deps:
           - ""
@@ -62,11 +63,11 @@ jobs:
         conda config --set channel_priority strict
         conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
-    - name: Install message-ix
-      if: ${{ matrix.os == 'windows-latest' }}
-      run: |
-        conda config --prepend channels conda-forge
-        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
+#    - name: Install message-ix
+#     if: ${{ matrix.os == 'windows-latest' }}
+#     run: |
+#        conda config --prepend channels conda-forge
+#        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -102,5 +102,7 @@ jobs:
         message-ix show-versions
         ixmp --platform default list
 
+        pip show message-ix
+
         conda list
-        pytest --pyargs message_ix.tests.test_core::test_add_cat --trace-config --full-trace -p ixmp.testing -p no:faulthandler
+        pytest --pyargs message_ix.tests.test_core::test_add_cat --trace-config -p ixmp.testing -p no:faulthandler

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -27,7 +27,7 @@ jobs:
       fail-fast: false
 
     runs-on: ${{ matrix.os }}
-    name: ${{ matrix.os }}-${{ matrix.conda }}
+    name: ${{ matrix.os }}-${{ matrix.conda }}-${{ matrix.extra_deps }}
 
     steps:
 #    - name: Cache Anaconda installer, conda packages
@@ -46,7 +46,7 @@ jobs:
       uses: iiasa/actions/setup-conda@main
       with:
         installer: ${{ matrix.conda }}
-        version: 2021.11
+        version: 2019.03
 
     - name: Setup miniconda
       if: ${{ matrix.conda == 'miniconda' }}
@@ -103,4 +103,4 @@ jobs:
         ixmp --platform default list
 
         conda list
-        pytest --pyargs message_ix.tests.test_core::test_add_cat -p ixmp.testing -p no:faulthandler
+        pytest --pyargs message_ix.tests.test_core::test_add_cat --trace-config --full-trace -p ixmp.testing -p no:faulthandler

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -60,7 +60,9 @@ jobs:
 
     - name: Configure Shell for "conda activate" macos-latest
       if: ${{ matrix.os == 'macos-latest' }}
-      run: conda init bash
+      run: |
+        conda init bash
+        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh
 
     - name: Configure PowerShell for "conda activate" windows-latest
       if: ${{ matrix.os == 'windows-latest' }}
@@ -82,7 +84,6 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
-        source ~/${{ matrix.conda }}3/etc/profile.d/conda.sh
         conda activate ${{ env.CONDA_ENV }}
 
         conda info

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -18,8 +18,8 @@ jobs:
   conda:
     strategy:
       matrix:
-        os: [windows-latest, macos-latest]
-#        os: [windows-latest]
+#        os: [windows-latest, macos-latest]
+        os: [macos-latest]
         conda: [anaconda, miniconda]
         extra_deps:
           - ""
@@ -82,6 +82,7 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
+        source ~/${{ matrix.conda }}3/etc/profile.d/conda.sh
         conda activate ${{ env.CONDA_ENV }}
 
         conda info

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -83,13 +83,13 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
-        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh || true
+        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh
         conda activate ${{ env.CONDA_ENV }}
 
         conda info
 
         message-ix show-versions
-        ixmp --platform default list
+        ixmp --platform default list || true
 
         conda list
         pytest --pyargs message_ix.tests.test_core::test_add_cat --rootdir=D:\a\_actions\iiasa\actions\main\setup-conda\${{ matrix.conda }}3\envs\testenv\lib\site-packages -p ixmp.testing -p no:faulthandler

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -92,4 +92,5 @@ jobs:
         ixmp --platform default list || true
 
         conda list
-        pytest --pyargs message_ix.tests.test_core::test_add_cat --rootdir=D:\a\_actions\iiasa\actions\main\setup-conda\${{ matrix.conda }}3\envs\testenv\lib\site-packages -p ixmp.testing -p no:faulthandler
+        pytest --pyargs message_ix.tests.test_core::test_add_cat --full-trace --trace-config --rootdir=/Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/envs/testenv/lib/python3.9/site-packages -p ixmp.testing -p no:faulthandler
+#        pytest --pyargs message_ix.tests.test_core::test_add_cat --rootdir=D:\a\_actions\iiasa\actions\main\setup-conda\${{ matrix.conda }}3\envs\testenv\lib\site-packages -p ixmp.testing -p no:faulthandler

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -30,16 +30,16 @@ jobs:
     name: ${{ matrix.os }}-${{ matrix.conda }}
 
     steps:
-    - name: Cache Anaconda installer, conda packages
-      uses: actions/cache@v2
-      with:
-        path: |
-          $CONDA/pkgs
-          ~/.conda/pkgs
-          ~/anaconda3/pkgs
-          ~/appdata/local/conda/conda/pkgs
-          ${{ github.workspace }}/Anaconda*.exe
-        key: ${{ matrix.os }}-${{ matrix.conda }}
+#    - name: Cache Anaconda installer, conda packages
+#      uses: actions/cache@v2
+#      with:
+#        path: |
+#          $CONDA/pkgs
+#          ~/.conda/pkgs
+#          ~/anaconda3/pkgs
+#          ~/appdata/local/conda/conda/pkgs
+#          ${{ github.workspace }}/Anaconda*.exe
+#        key: ${{ matrix.os }}-${{ matrix.conda }}
 
     - name: Setup anaconda
       if: ${{ matrix.conda == 'anaconda' }}

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -61,8 +61,17 @@ jobs:
       run: conda info
 
     - name: Configure Shell for "conda activate"
-      if: ${{ matrix.os == 'macos-latest' }}
-      run: conda init bash
+      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda == 'miniconda' }}
+      run: |
+        conda init bash
+        source ~/Miniconda3/etc/profile.d/conda.sh
+
+
+    - name: Configure Shell for "conda activate"
+      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda == 'anaconda' }}
+      run: |
+        conda init bash
+        source ~/Anaconda3/etc/profile.d/conda.csh
 
 #    - name: Configure PowerShell for "conda activate"
 #      if: ${{ matrix.os == 'windows-latest' }}

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -87,8 +87,8 @@ jobs:
 #        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
     - name: Install message-ix
-     if: ${{ matrix.os == 'windows-latest' }}
-     run: |
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: |
         conda config --prepend channels conda-forge
         conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -39,42 +39,34 @@ jobs:
           ${{ github.workspace }}/Anaconda*.*
         key: ${{ matrix.os }}-${{ matrix.conda_type }}
 
-    - name: Download and install Anaconda for Windows; add to PATH
-      if: ${{ matrix.os == 'windows-latest' }} && ${{ matrix.conda_type == 'anaconda' }}
-      working-directory: ${{ github.workspace }}
-      run: |
-        curl -O https://repo.anaconda.com/archive/Anaconda3-2021.05-Windows-x86_64.exe
-        Start-Process "Anaconda3-2021.05-Windows-x86_64.exe" "/S", "/D=${env:GITHUB_WORKSPACE}\anaconda3" -Wait
-        "${env:GITHUB_WORKSPACE}/anaconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
+    - name: Setup anaconda
+      if: ${{ matrix.conda == 'anaconda' }}
+      # Todo change branch back to `main`
+      uses: iiasa/actions/setup-conda@feature/setup-conda-action
+      with:
+        installer: ${{ matrix.conda }}
+        version: 2019.03
 
-    - name: Download and install Miniconda for Windows; add to PATH
-      if: ${{ matrix.os == 'windows-latest' }} && ${{ matrix.conda_type == 'miniconda' }}
-      working-directory: ${{ github.workspace }}
-      run: |
-        curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe
-        Start-Process "Miniconda3-latest-Windows-x86_64.exe" "/S", "/D=${env:GITHUB_WORKSPACE}\miniconda3" -Wait
-        "${env:GITHUB_WORKSPACE}/miniconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
-
-    - name: Download and install Anaconda for MacOS; add to PATH
-      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda_type == 'anaconda' }}
-      working-directory: ${{ github.workspace }}
-      run: |
-        curl -O https://repo.anaconda.com/archive/Anaconda3-2021.05-MacOSX-x86_64.pkg
-        Start-Process "Anaconda3-2021.05-MacOSX-x86_64.pkg" "/S", "/D=${env:GITHUB_WORKSPACE}\anaconda3" -Wait
-        "${env:GITHUB_WORKSPACE}/anaconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
-
-    - name: Download and install Miniconda for MacOS; add to PATH
-      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda_type == 'miniconda' }}
-      working-directory: ${{ github.workspace }}
-      run: |
-        curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.pkg
-        Start-Process "Miniconda3-latest-MacOSX-x86_64.pkg" "/S", "/D=${env:GITHUB_WORKSPACE}\miniconda3" -Wait
-        "${env:GITHUB_WORKSPACE}/miniconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
+    - name: Setup miniconda
+      if: ${{ matrix.conda == 'miniconda' }}
+      # Todo change branch back to `main`
+      uses: iiasa/actions/setup-conda@feature/setup-conda-action
+      with:
+        installer: ${{ matrix.conda }}
+        version: py39_4.9.2
 
     - name: Configure PowerShell for "conda activate"
       run: conda init powershell
 
     - name: Install message-ix
+      if: ${{ matrix.os == 'macos-latest' }}
+      run: |
+        conda config --prepend channels conda-forge
+        conda config --set channel_priority strict
+        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
+
+    - name: Install message-ix
+      if: ${{ matrix.os == 'windows-latest' }}
       run: |
         conda config --prepend channels conda-forge
         conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -18,8 +18,8 @@ jobs:
   conda:
     strategy:
       matrix:
-        # os: [windows-latest, macos-latest]
-        os: [windows-latest]
+        os: [windows-latest, macos-latest]
+#        os: [windows-latest]
         conda: [anaconda, miniconda]
         extra_deps:
           - ""
@@ -58,35 +58,22 @@ jobs:
     - name: Check conda version and location
       run: conda info
 
-#    - name: Configure Shell for "conda activate"
-#      if: |
-#        ${{ matrix.os == 'macos-latest' }} ||
-#        ${{ matrix.conda == 'miniconda' }}
-#      run: |
-#        conda init bash
-#        source ~/Miniconda3/etc/profile.d/conda.sh
-#
-#
-#    - name: Configure Shell for "conda activate"
-#      if: |
-#        ${{ matrix.os == 'macos-latest' }} ||
-#        ${{ matrix.conda == 'anaconda' }}
-#      run: |
-#        conda init bash
-#        source ~/Anaconda3/etc/profile.d/conda.csh
+    - name: Configure Shell for "conda activate" macos-latest
+      if: ${{ matrix.os == 'macos-latest' }}
+      run: conda init bash
 
-    - name: Configure PowerShell for "conda activate"
+    - name: Configure PowerShell for "conda activate" windows-latest
       if: ${{ matrix.os == 'windows-latest' }}
       run: conda init powershell
-#
-#    - name: Install message-ix
-#      if: ${{ matrix.os == 'macos-latest' }}
-#      run: |
-#        conda config --prepend channels conda-forge
-#        conda config --set channel_priority strict
-#        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
-    - name: Install message-ix
+    - name: Install message-ix on macos
+      if: ${{ matrix.os == 'macos-latest' }}
+      run: |
+        conda config --prepend channels conda-forge
+        conda config --set channel_priority strict
+        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
+
+    - name: Install message-ix on windows
       if: ${{ matrix.os == 'windows-latest' }}
       run: |
         conda config --prepend channels conda-forge
@@ -102,7 +89,5 @@ jobs:
         message-ix show-versions
         ixmp --platform default list
 
-        pip show message-ix
-
         conda list
-        pytest --pyargs message_ix.tests.test_core::test_add_cat --trace-config -p ixmp.testing -p no:faulthandler
+        pytest --pyargs message_ix.tests.test_core::test_add_cat --rootdir=D:\a\_actions\iiasa\actions\main\setup-conda\${{ matrix.conda }}3\envs\testenv\lib\site-packages -p ixmp.testing -p no:faulthandler

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -73,7 +73,6 @@ jobs:
       if: ${{ matrix.os == 'macos-latest' }}
       run: |
         conda config --prepend channels conda-forge
-        conda config --set channel_priority strict
         conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
     - name: Install message-ix on windows

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -61,14 +61,18 @@ jobs:
       run: conda info
 
     - name: Configure Shell for "conda activate"
-      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda == 'miniconda' }}
+      if: |
+        ${{ matrix.os == 'macos-latest' }} ||
+        ${{ matrix.conda == 'miniconda' }}
       run: |
         conda init bash
         source ~/Miniconda3/etc/profile.d/conda.sh
 
 
     - name: Configure Shell for "conda activate"
-      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda == 'anaconda' }}
+      if: |
+        ${{ matrix.os == 'macos-latest' }} ||
+        ${{ matrix.conda == 'anaconda' }}
       run: |
         conda init bash
         source ~/Anaconda3/etc/profile.d/conda.csh

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -24,9 +24,10 @@ jobs:
         extra_deps:
           - ""
           - "JPype1=1.2.1"
+      fail-fast: false
 
     runs-on: ${{ matrix.os }}
-    name: ${{ matrix.conda }}
+    name: ${{ matrix.os }}-${{ matrix.conda }}
 
     steps:
     - name: Cache Anaconda installer, conda packages
@@ -55,6 +56,9 @@ jobs:
       with:
         installer: ${{ matrix.conda }}
         version: py39_4.9.2
+
+    - name: Check conda version and location
+      run: conda info
 
     - name: Configure Shell for "conda activate"
       if: ${{ matrix.os == 'macos-latest' }}

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -63,7 +63,6 @@ jobs:
       if: ${{ matrix.os == 'macos-latest' }}
       run: |
         conda init bash
-        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh
 
     - name: Configure PowerShell for "conda activate" windows-latest
       if: ${{ matrix.os == 'windows-latest' }}
@@ -84,6 +83,7 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
+        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh
         conda activate ${{ env.CONDA_ENV }}
 
         conda info

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -24,7 +24,7 @@ jobs:
         conda: [miniconda]
         extra_deps:
           - ""
-          - "JPype1=1.2.1"
+#          - "JPype1=1.2.1"
       fail-fast: false
 
     runs-on: ${{ matrix.os }}
@@ -84,9 +84,7 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
-        ps -p $$
-
-        source activate ${{ env.CONDA_ENV }}
+        conda activate ${{ env.CONDA_ENV }}
 
         conda info
 

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -7,8 +7,8 @@ on:
   schedule:
   - cron: "0 5 * * *"
   # Uncomment to debug
-  # pull_request:
-  #   branches: [ main ]
+  pull_request:
+   branches: [ main ]
 
 env:
   # Name of the test environment to create

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -19,7 +19,7 @@ jobs:
     strategy:
       matrix:
         # os: [windows-latest, macos-latest]
-        os: [macos-latest]
+        os: [windows-latest]
         conda: [anaconda, miniconda]
         extra_deps:
           - ""
@@ -30,69 +30,67 @@ jobs:
     name: ${{ matrix.os }}-${{ matrix.conda }}
 
     steps:
-    - name: Cache Anaconda installer, conda packages
-      uses: actions/cache@v2
-      with:
-        path: |
-          $CONDA/pkgs
-          ~/.conda/pkgs
-          ~/anaconda3/pkgs
-          ~/appdata/local/conda/conda/pkgs
-          ${{ github.workspace }}/Anaconda*.*
-        key: ${{ matrix.os }}-${{ matrix.conda }}
+#    - name: Cache Anaconda installer, conda packages
+#      uses: actions/cache@v2
+#      with:
+#        path: |
+#          $CONDA/pkgs
+#          ~/.conda/pkgs
+#          ~/anaconda3/pkgs
+#          ~/appdata/local/conda/conda/pkgs
+#          ${{ github.workspace }}/Anaconda*.*
+#        key: ${{ matrix.os }}-${{ matrix.conda }}
 
     - name: Setup anaconda
       if: ${{ matrix.conda == 'anaconda' }}
-      # Todo change branch back to `main`
-      uses: iiasa/actions/setup-conda@feature/setup-conda-action
+      uses: iiasa/actions/setup-conda@main
       with:
         installer: ${{ matrix.conda }}
-        version: 2019.03
+        version: 2021.11
 
     - name: Setup miniconda
       if: ${{ matrix.conda == 'miniconda' }}
-      # Todo change branch back to `main`
-      uses: iiasa/actions/setup-conda@feature/setup-conda-action
+      uses: iiasa/actions/setup-conda@main
       with:
         installer: ${{ matrix.conda }}
-        version: py39_4.9.2
+        version: py39_4.10.3
 
     - name: Check conda version and location
       run: conda info
 
-    - name: Configure Shell for "conda activate"
-      if: |
-        ${{ matrix.os == 'macos-latest' }} ||
-        ${{ matrix.conda == 'miniconda' }}
-      run: |
-        conda init bash
-        source ~/Miniconda3/etc/profile.d/conda.sh
+#    - name: Configure Shell for "conda activate"
+#      if: |
+#        ${{ matrix.os == 'macos-latest' }} ||
+#        ${{ matrix.conda == 'miniconda' }}
+#      run: |
+#        conda init bash
+#        source ~/Miniconda3/etc/profile.d/conda.sh
+#
+#
+#    - name: Configure Shell for "conda activate"
+#      if: |
+#        ${{ matrix.os == 'macos-latest' }} ||
+#        ${{ matrix.conda == 'anaconda' }}
+#      run: |
+#        conda init bash
+#        source ~/Anaconda3/etc/profile.d/conda.csh
 
-
-    - name: Configure Shell for "conda activate"
-      if: |
-        ${{ matrix.os == 'macos-latest' }} ||
-        ${{ matrix.conda == 'anaconda' }}
-      run: |
-        conda init bash
-        source ~/Anaconda3/etc/profile.d/conda.csh
-
-#    - name: Configure PowerShell for "conda activate"
-#      if: ${{ matrix.os == 'windows-latest' }}
-#      run: conda init powershell
+    - name: Configure PowerShell for "conda activate"
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: conda init powershell
+#
+#    - name: Install message-ix
+#      if: ${{ matrix.os == 'macos-latest' }}
+#      run: |
+#        conda config --prepend channels conda-forge
+#        conda config --set channel_priority strict
+#        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
     - name: Install message-ix
-      if: ${{ matrix.os == 'macos-latest' }}
-      run: |
+     if: ${{ matrix.os == 'windows-latest' }}
+     run: |
         conda config --prepend channels conda-forge
-        conda config --set channel_priority strict
         conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
-
-#    - name: Install message-ix
-#     if: ${{ matrix.os == 'windows-latest' }}
-#     run: |
-#        conda config --prepend channels conda-forge
-#        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -30,16 +30,16 @@ jobs:
     name: ${{ matrix.os }}-${{ matrix.conda }}
 
     steps:
-#    - name: Cache Anaconda installer, conda packages
-#      uses: actions/cache@v2
-#      with:
-#        path: |
-#          $CONDA/pkgs
-#          ~/.conda/pkgs
-#          ~/anaconda3/pkgs
-#          ~/appdata/local/conda/conda/pkgs
-#          ${{ github.workspace }}/Anaconda*.*
-#        key: ${{ matrix.os }}-${{ matrix.conda }}
+    - name: Cache Anaconda installer, conda packages
+      uses: actions/cache@v2
+      with:
+        path: |
+          $CONDA/pkgs
+          ~/.conda/pkgs
+          ~/anaconda3/pkgs
+          ~/appdata/local/conda/conda/pkgs
+          ${{ github.workspace }}/Anaconda*.exe
+        key: ${{ matrix.os }}-${{ matrix.conda }}
 
     - name: Setup anaconda
       if: ${{ matrix.conda == 'anaconda' }}

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -56,8 +56,13 @@ jobs:
         installer: ${{ matrix.conda }}
         version: py39_4.9.2
 
-    - name: Configure PowerShell for "conda activate"
-      run: conda init powershell
+    - name: Configure Shell for "conda activate"
+      if: ${{ matrix.os == 'macos-latest' }}
+      run: conda init bash
+
+#    - name: Configure PowerShell for "conda activate"
+#      if: ${{ matrix.os == 'windows-latest' }}
+#      run: conda init powershell
 
     - name: Install message-ix
       if: ${{ matrix.os == 'macos-latest' }}

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -85,7 +85,9 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
-        conda activate ${{ env.CONDA_ENV }}
+        ps -p $$
+
+        source activate ${{ env.CONDA_ENV }}
 
         conda info
 

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -19,13 +19,13 @@ jobs:
     strategy:
       matrix:
         os: [windows-latest, macos-latest]
-        conda_type: [anaconda, miniconda]
+        conda: [anaconda, miniconda]
         extra_deps:
           - ""
           - "JPype1=1.2.1"
 
     runs-on: ${{ matrix.os }}
-    name: ${{ matrix.conda_type }}
+    name: ${{ matrix.conda }}
 
     steps:
     - name: Cache Anaconda installer, conda packages
@@ -37,7 +37,7 @@ jobs:
           ~/anaconda3/pkgs
           ~/appdata/local/conda/conda/pkgs
           ${{ github.workspace }}/Anaconda*.*
-        key: ${{ matrix.os }}-${{ matrix.conda_type }}
+        key: ${{ matrix.os }}-${{ matrix.conda }}
 
     - name: Setup anaconda
       if: ${{ matrix.conda == 'anaconda' }}

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -62,7 +62,6 @@ jobs:
     - name: Install message-ix
       run: |
         conda config --prepend channels conda-forge
-        conda config --set channel_priority strict
         conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
     - name: Check

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -16,15 +16,15 @@ env:
 
 jobs:
   conda:
-    runs-on: windows-latest
-
     strategy:
       matrix:
+        os: [windows-latest, macos-latest]
         conda_type: [anaconda, miniconda]
         extra_deps:
           - ""
           - "JPype1=1.2.1"
 
+    runs-on: ${{ matrix.os }}
     name: ${{ matrix.conda_type }}
 
     steps:
@@ -36,25 +36,40 @@ jobs:
           ~/.conda/pkgs
           ~/anaconda3/pkgs
           ~/appdata/local/conda/conda/pkgs
-          ${{ github.workspace }}/Anaconda*.exe
-        key: windows-latest-${{ matrix.conda_type }}
+          ${{ github.workspace }}/Anaconda*.*
+        key: ${{ matrix.os }}-${{ matrix.conda_type }}
 
-    - name: Download and install Anaconda; add to PATH
-      if: ${{ matrix.conda_type == 'anaconda' }}
+    - name: Download and install Anaconda for Windows; add to PATH
+      if: ${{ matrix.os == 'windows-latest' }} && ${{ matrix.conda_type == 'anaconda' }}
       working-directory: ${{ github.workspace }}
       run: |
         curl -O https://repo.anaconda.com/archive/Anaconda3-2021.05-Windows-x86_64.exe
         Start-Process "Anaconda3-2021.05-Windows-x86_64.exe" "/S", "/D=${env:GITHUB_WORKSPACE}\anaconda3" -Wait
         "${env:GITHUB_WORKSPACE}/anaconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
 
-    - name: Download and install Miniconda; add to PATH
-      if: ${{ matrix.conda_type == 'miniconda' }}
+    - name: Download and install Miniconda for Windows; add to PATH
+      if: ${{ matrix.os == 'windows-latest' }} && ${{ matrix.conda_type == 'miniconda' }}
       working-directory: ${{ github.workspace }}
       run: |
         curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe
         Start-Process "Miniconda3-latest-Windows-x86_64.exe" "/S", "/D=${env:GITHUB_WORKSPACE}\miniconda3" -Wait
         "${env:GITHUB_WORKSPACE}/miniconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
 
+    - name: Download and install Anaconda for MacOS; add to PATH
+      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda_type == 'anaconda' }}
+      working-directory: ${{ github.workspace }}
+      run: |
+        curl -O https://repo.anaconda.com/archive/Anaconda3-2021.05-MacOSX-x86_64.pkg
+        Start-Process "Anaconda3-2021.05-MacOSX-x86_64.pkg" "/S", "/D=${env:GITHUB_WORKSPACE}\anaconda3" -Wait
+        "${env:GITHUB_WORKSPACE}/anaconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
+
+    - name: Download and install Miniconda for MacOS; add to PATH
+      if: ${{ matrix.os == 'macos-latest' }} && ${{ matrix.conda_type == 'miniconda' }}
+      working-directory: ${{ github.workspace }}
+      run: |
+        curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.pkg
+        Start-Process "Miniconda3-latest-MacOSX-x86_64.pkg" "/S", "/D=${env:GITHUB_WORKSPACE}\miniconda3" -Wait
+        "${env:GITHUB_WORKSPACE}/miniconda3/Scripts" | Out-File -FilePath $env:GITHUB_PATH -Append
 
     - name: Configure PowerShell for "conda activate"
       run: conda init powershell

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -7,8 +7,8 @@ on:
   schedule:
   - cron: "0 5 * * *"
   # Uncomment to debug
-  pull_request:
-   branches: [ main ]
+  # pull_request:
+  #   branches: [ main ]
 
 jobs:
   conda:

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -20,7 +20,8 @@ jobs:
       matrix:
 #        os: [windows-latest, macos-latest]
         os: [macos-latest]
-        conda: [anaconda, miniconda]
+#        conda: [anaconda, miniconda]
+        conda: [miniconda]
         extra_deps:
           - ""
           - "JPype1=1.2.1"

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -55,9 +55,6 @@ jobs:
         installer: ${{ matrix.conda }}
         version: py39_4.9.2
 
-    - name: Configure PowerShell for "conda activate"
-      run: conda init powershell
-
     - name: Install message-ix
       if: ${{ matrix.os == 'macos-latest' }}
       run: |

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -10,87 +10,79 @@ on:
   pull_request:
    branches: [ main ]
 
-env:
-  # Name of the test environment to create
-  CONDA_ENV: testenv
-
 jobs:
   conda:
     strategy:
       matrix:
-#        os: [windows-latest, macos-latest]
-        os: [macos-latest]
-#        conda: [anaconda, miniconda]
-        conda: [miniconda]
-        extra_deps:
-          - ""
-#          - "JPype1=1.2.1"
+        os: [windows-latest, macos-latest]
+        conda:
+        - {installer: anaconda, version: 2019.03}
+        - {installer: miniconda, version: py39_4.10.3}
+        extra-deps: ["", "JPype1=1.2.1"]
       fail-fast: false
 
     runs-on: ${{ matrix.os }}
-    name: ${{ matrix.os }}-${{ matrix.conda }}-${{ matrix.extra_deps }}
+    name: ${{ matrix.os }}-${{ matrix.conda.installer }}-${{ matrix.extra-deps }}
 
     steps:
-#    - name: Cache Anaconda installer, conda packages
-#      uses: actions/cache@v2
-#      with:
-#        path: |
-#          $CONDA/pkgs
-#          ~/.conda/pkgs
-#          ~/anaconda3/pkgs
-#          ~/appdata/local/conda/conda/pkgs
-#          ${{ github.workspace }}/Anaconda*.exe
-#        key: ${{ matrix.os }}-${{ matrix.conda }}
+    - name: Cache Anaconda installer, conda packages
+      uses: actions/cache@v2
+      with:
+        path: |
+          $CONDA/pkgs
+          ~/.conda/pkgs
+          ~/anaconda3/pkgs
+          ~/appdata/local/conda/conda/pkgs
+          ${{ github.workspace }}/Anaconda*.exe
+        key: ${{ matrix.os }}-${{ matrix.conda.installer }}
 
-    - name: Setup anaconda
-      if: ${{ matrix.conda == 'anaconda' }}
+    - name: Setup Anaconda or miniconda
       uses: iiasa/actions/setup-conda@main
       with:
-        installer: ${{ matrix.conda }}
-        version: 2019.03
+        installer: ${{ matrix.conda.installer }}
+        version: ${{ matrix.conda.version }}
 
-    - name: Setup miniconda
-      if: ${{ matrix.conda == 'miniconda' }}
-      uses: iiasa/actions/setup-conda@main
-      with:
-        installer: ${{ matrix.conda }}
-        version: py39_4.10.3
-
-    - name: Check conda version and location
-      run: conda info
-
-    - name: Configure Shell for "conda activate" macos-latest
-      if: ${{ matrix.os == 'macos-latest' }}
+    # TODO move the following 2 steps into the above action
+    - name: Determine shell on current OS
+      id: shell
       run: |
-        conda init bash
+        init, profile = {
+          "macos": ("bash", "source ~/.bash_profile"),
+          "windows": ("powershell", ""),
+        }.get("${{ matrix.os }}".split("-")[0])
+        print(f"""
+          ::set-output name=init::{init}
+          ::set-output name=profile::{profile}
+        """)
+      shell: python
 
-    - name: Configure PowerShell for "conda activate" windows-latest
-      if: ${{ matrix.os == 'windows-latest' }}
-      run: conda init powershell
+    - name: Initialize shell for "conda activate"
+      run: conda init ${{ steps.shell.outputs.init }}
 
-    - name: Install message-ix on macos
-      if: ${{ matrix.os == 'macos-latest' }}
+    - name: Configure conda channels, create environment, and install message-ix
       run: |
         conda config --prepend channels conda-forge
-        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
 
-    - name: Install message-ix on windows
-      if: ${{ matrix.os == 'windows-latest' }}
+        # Also install pytest and packages required for testing. pip/PyPI
+        # supports this via message_ix[testing], but conda does not.
+        conda create --quiet --name testenv message-ix ${{ matrix.extra-deps }} pytest asyncssh sphinx
+        conda list --name testenv
+
+    - name: Check CLI commands and run test
       run: |
-        conda config --prepend channels conda-forge
-        conda create --name ${{ env.CONDA_ENV }} --quiet message-ix pytest ${{ matrix.extra_deps }}
+        # Source profile. On non-windows OSes, GHA invokes bash with
+        # "--noprofile", so the file written by "conda init bash" above is not
+        # read automatically
+        ${{ steps.shell.outputs.profile }}
 
-    - name: Check
-      # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
-      run: |
-        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh
-        conda activate ${{ env.CONDA_ENV }}
-
+        # Activate the test environment
+        conda activate testenv
         conda info
 
+        # Check CLI tools
         message-ix show-versions
         ixmp --platform default list || true
 
-        conda list
-        pytest --pyargs message_ix.tests.test_core::test_add_cat --full-trace --trace-config --rootdir=/Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/envs/testenv/lib/python3.9/site-packages -p ixmp.testing -p no:faulthandler
-#        pytest --pyargs message_ix.tests.test_core::test_add_cat --rootdir=D:\a\_actions\iiasa\actions\main\setup-conda\${{ matrix.conda }}3\envs\testenv\lib\site-packages -p ixmp.testing -p no:faulthandler
+        # Run a single test from the test suite to touch JDBCBackend Java code,
+        # via JPype
+        pytest --pyargs message_ix -p ixmp.testing -p no:faulthandler -k "test_add_cat" --verbose -rA --color=yes

--- a/.github/workflows/conda.yaml
+++ b/.github/workflows/conda.yaml
@@ -83,7 +83,7 @@ jobs:
     - name: Check
       # Check environment and run a single test from the suite to test code path to ixmp JDBCBackend
       run: |
-        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh  || true
+        source /Users/runner/work/_actions/iiasa/actions/main/setup-conda/${{ matrix.conda }}3/etc/profile.d/conda.sh || true
         conda activate ${{ env.CONDA_ENV }}
 
         conda info

--- a/message_ix/tests/test_tutorials.py
+++ b/message_ix/tests/test_tutorials.py
@@ -1,3 +1,4 @@
+import os
 import sys
 from shutil import copyfile
 from typing import List, Tuple
@@ -88,6 +89,10 @@ def nb_path(request, tutorial_path):
 
 # Parametrize the first 3 arguments using the variables *tutorial* and *ids*.
 # Argument 'nb_path' is indirect so that the above function can modify it.
+@pytest.mark.xfail(
+    condition="GITHUB_ACTIONS" in os.environ and sys.platform == "darwin",
+    reason="Flaky; CellTimeoutError occurs on first executed cell",
+)
 @pytest.mark.parametrize(
     "nb_path,cell_values,run_args", tutorials, ids=ids, indirect=["nb_path"]
 )

